### PR TITLE
[8.0][ADD] web_form_sticky_header

### PR DESCRIPTION
**Pull request by daramousk**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Fixed = sticky?

Any screenshot?

**Comment by daramousk** [comment]
@pedrobaeza Yes, the header is static, attached a screenshot

**Comment by pedrobaeza** [comment]
I prefer then the name `web_form_sticky_header`.

**Comment by daramousk** [comment]
@pedrobaeza Renamed the module

File: web_form_sticky_header/static/src/js/web_form_sticky_header.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+//-*- coding: utf-8 -*-
```
Comment (NL66278) [inline web_form_sticky_header/static/src/js/web_form_sticky_header.js]: This is only for python files, not javascript

Comment (daramousk) [reply web_form_sticky_header/static/src/js/web_form_sticky_header.js]: @NL66278 Yes wrong vim keymap, I will reimplement this module

File: web_form_sticky_header/static/src/js/web_form_sticky_header.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+openerp.web_form_sticky_header = function(instance)
+{
+    instance.web.FormView.include({
+        start: function() {
+            self = this;
+            return this._super().then(function() {
+                var header = self.$el.find('header')
+                header.css({'background-color': '#ededed;'})
+                header.insertAfter(self.$el.parents().children('.oe_view_manager_header'))
```
Comment (NL66278) [inline web_form_sticky_header/static/src/js/web_form_sticky_header.js]: Please give some comments here on what is happening here, and how this leads to a fixed header.

**Comment by daramousk** [comment]
@NL66278 Reimplemented using CSS.

File: web_form_sticky_header/static/src/css/sticky_header.css [hunk 3]
```diff
@@ -0,0 +1,8 @@
+.oe_form header {
```
Comment (NL66278) [inline web_form_sticky_header/static/src/css/sticky_header.css:1]: The original specifies it must be within an openerp class. Should that not be also done for this one?

**Review by NL66278 — APPROVED** [review]
👍 LGTM and tested. One remark to look at, but not critical. Big leap in usability!

**Comment by daramousk** [comment]
@NL66278 This still has one issue, if you click on the More menu the bar overlaps because of the z-index, will need to find a way around this

**Comment by NL66278** [comment]
And it is the same problem for print and attachment menu's. So calls for
some general solution.

File: web_form_sticky_header/views/templates.xml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<openerp>
+    <data>
+        <template id="assets_backend" name="web_form_sticky_header assets" inherit_id="web.assets_backend">
+            <xpath expr="." position="inside">
```
Comment (gfcapalbo) [inline web_form_sticky_header/views/templates.xml:5]: according to what we discussed today tthis would place it last. equivalent to  expr="//link[@href=/base/static/src/css/modules.css]"   , you are injecting in https://github.com/OCA/OCB/blob/8.0/addons/web/views/webclient_templates.xml#L90 .   looks ok.

File: web_form_sticky_header/static/src/css/sticky_header.css (hunk 3 above)
Comment (gfcapalbo) [reply web_form_sticky_header/static/src/css/sticky_header.css:1]: https://github.com/OCA/OCB/blob/8.0/addons/web/static/src/css/base.css#L1922

I believe ronald is right here

change with .openerp .oe_form header {

File: web_form_sticky_header/static/src/css/sticky_header.css [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+    position: sticky;
```
Comment (hbrunn) [inline web_form_sticky_header/static/src/css/sticky_header.css:2]: I was not so sure about this one, but https://caniuse.com/#search=sticky says this is a thing. TIL, thanks!

**Comment by daramousk** [comment]
@hbrunn @gfcapalbo @NL66278  Everything should work perfectly now

**Review by hbrunn — APPROVED** [review]
nice!

In the chat, I also proposed to add

```css
div.modal-header {
    position: sticky;
    bottom: 0px;
    background: white;
    right: 0px;
    left: 0px;
}
```

to have a similar function for big popup windows.

but as that's a feature request, take or ignore at your convenience

**Comment by hbrunn** [comment]
PS: don't forget to rebase

**Comment by lfreeke** [comment]
:+1: (test)

**Review by yajo — APPROVED** [review]
Code OK

**Comment by yajo** [comment]
I feel this will harm phone users, but since it's already unusable, merging...

**Pull request merged**